packages/store/src/internal/state-factory.ts

```diff
@@ -51,10 +49,16 @@ export class StateFactory {
     private _stateContextFactory: StateContextFactory
   ) {}
 
+  private _states: MappedStore[] = [];
+
+  public get states(): MappedStore[] {
+    return this._parentFactory ? this._parentFactory.states : this._states;
+  }
+
   /**
    * Add a new state to the global defs.
    */
-  add(oneOrManyStateClasses: StateClass | StateClass[]): MappedStore[] {
+  public add(oneOrManyStateClasses: StateClass | StateClass[]): MappedStore[] {
```

Comment:
I want to start sticking throughout the project so that we are in java style, because, it teaches us to think in advance about what methods are used in the outside class.

Further, at the tslint level, we will include warnings that you should use access modifiers.

Reply:
I do like that all the methods are aligned when you do need private methods.
But this is the only reason, at the end it's a matter of preference.

Reply:
I mostly write to Java and understand why it’s convenient
I think it's time for us to get used to writing modifiers, because it teaches good code.

Reply:
Agree, it shouldn't be in this PR.
We should talk about conventions like this in the slack channel.